### ISSUE TITLE
[Bug] Equations should be reset on each document render

### ISSUE THREAD

**Issue by Mister-Hope** [mdit-plugins/mdit-plugins#25]
### Package name

@md-it/plugin-mathjax

### Describe the bug

When calling `mardownIt.render` on different markdown documents with equation label, errors like this will be shown:

```
Label 'eq:m1' multiply defined
```

So the plugin needs a new api to switch to a new page while preserving style cache.

Related issues:

- https://github.com/vuepress-theme-hope/vuepress-theme-hope/issues/2790
- https://github.com/mathjax/MathJax/issues/2931

Related docs:

- https://docs.mathjax.org/en/latest/web/typeset.html
